### ISSUE TITLE
Contradictory indentation errors with multiline type parameter list

### ISSUE THREAD

**Issue by bethcutler** [pinterest/ktlint#680]
If a code snippet contains a multi-line type parameter list, ktlint may produce contradictory errors with the experimental indentation rule.

E.g. for this code snippet
```kotlin
object TestCase {
  inline fun <
    T1,
    T2,
    T3> create(
      t1: T1,
      t2: T2,
      t3: T3
  ) {
    // do things
  }
}
```

... ktlint produces this errors for each function parameter line (t1, t2, t3):
> Unexpected indentation (6) (should be 4)

If the indentation on the function parameter lines is "corrected" like so
```kotlin
object TestCase {
  inline fun <
    T1,
    T2,
    T3> create(
    t1: T1,
    t2: T2,
    t3: T3
  ) {
    // do things
  }
}
```

... ktlint is still not satisfied, now producing this error on each of those lines:
> Unexpected indentation (expected 6, actual 4)

**Comment by Tapchicoma** [comment]
Probably could be covered by https://github.com/pinterest/ktlint/pull/587

**Comment by AdamMTGreenberg** [comment]
@Tapchicoma Added PR: https://github.com/pinterest/ktlint/pull/804 to enforce this behavior. However, it is unclear if we want to make this an enforced rule:

```
object TestCase {
  inline fun <
    T1,
    T2,
    T3> create(
    t1: T1,
    t2: T2,
    t3: T3
  ) {
    // do things
  }
}
```
to
```
object TestCase {
  inline fun <
    T1,
    T2,
    T3> create(
      t1: T1,
      t2: T2,
      t3: T3
  ) {
    // do things
  }
}
```
since the default spacing seems to leave the indentation flat. 

If we want to leave it as is, then I think we can close out this issue.
